clinical-domain-agent/src/test/java/care/smith/fts/cda/impl/TCACohortSelectorIT.java

```diff
@@ -0,0 +1,211 @@
+
+  @Override
+  protected Stream<TestStep<?>> createTestSteps() {
+    return Stream.of(
+        new TestStep<ConsentedPatient>() {
+          @Override
+          public MappingBuilder getBuilder() {
+            return TCACohortSelectorIT.getBuilderFetchAll();
+          }
+
+          @Override
+          public Flux<ConsentedPatient> executeStep() {
+            return TCACohortSelectorIT.cohortSelector.selectCohort(List.of());
+          }
+        },
+        new TestStep<ConsentedPatient>() {
+          @Override
+          public MappingBuilder getBuilder() {
+            return TCACohortSelectorIT.getBuilderFetch();
+          }
+
+          @Override
+          public Flux<ConsentedPatient> executeStep() {
+            return TCACohortSelectorIT.cohortSelector.selectCohort(List.of("id"));
+          }
+
+          @Override
+          public String acceptedContentType() {
+            return APPLICATION_JSON_VALUE;
+          }
+        });
+  }
+
+  @Test
+  void responseInvalidErrors() {
+    wireMock.register(getBuilderFetchAll().willReturn(ok().withBody("invalid")));
+    create(cohortSelector.selectCohort(List.of())).expectError().verify();
+  }
+
+  @Test
+  void badRequestErrors() throws JsonProcessingException {
+    var om = new ObjectMapper();
+    wireMock.register(
+        getBuilderFetchAll()
+            .willReturn(
+                badRequest()
+                    .withHeader("Content-Type", APPLICATION_JSON_VALUE)
+                    .withBody(
+                        om.writeValueAsString(
+                            ProblemDetail.forStatusAndDetail(
+                                BAD_REQUEST, "TCA Returns Bad Request")))));
+
+    create(cohortSelector.selectCohort(List.of()))
+        .expectErrorMessage("TCA Returns Bad Request")
+        .verify();
+  }
+
+  @Test
+  void consentBundleSucceeds() {
+    allCohortSelector.consentForOnePatient("patient");
+    create(cohortSelector.selectCohort(List.of())).expectNextCount(1).verifyComplete();
+  }
+
+  @Test
+  void consentBundleForIdsSucceeds() {
+    listCohortSelector.consentForOnePatient("patient");
+    create(cohortSelector.selectCohort(List.of("patient0"))).expectNextCount(1).verifyComplete();
+  }
+
+  @Test
+  void emptyOuterBundleGivesEmptyResult() {
+    var outer = Stream.<Resource>of().collect(toBundle());
+
+    wireMock.register(
+        getBuilderFetchAll()
+            .willReturn(
+                ok().withHeader(CONTENT_TYPE, APPLICATION_FHIR_JSON)
+                    .withBody(fhirResourceToString(outer))));
+    create(cohortSelector.selectCohort(List.of())).verifyComplete();
+  }
+
+  @Test
+  void emptyInnerBundleGivesEmptyResult() {
+    var inner = Stream.of(Stream.<Resource>of().collect(toBundle())).collect(toBundle());
+    wireMock.register(
+        getBuilderFetchAll()
+            .willReturn(
+                ok().withHeader(CONTENT_TYPE, APPLICATION_FHIR_JSON)
+                    .withBody(fhirResourceToString(inner))));
```

Comment:
We have fhirResponse in MockServerUtil, it does this. We could add one without status code, defaulting to OK.